### DISCUSSION TITLE
Calendar use Sunday as first weekday

### DISCUSSION THREAD

**Discussion by ReinforceZwei** [TriliumNext/Trilium#4540] [Show and tell]
## Sunday as First Weekday
Trilium calendar use Monday as first weekday by default and there is no option to change it. So I've made a frontend script to modify the calendar to make it use Sunday as first weekday.

https://gist.github.com/ReinforceZwei/af3254b0282e74f74addf2a73b4636a9

## How to use
1. Create a new note with type `JS Frontend`
2. Paste the code from the above gist
3. Add attribute `#run=frontendStartup` to enable the script

**Comment by TheBig-O** [comment]
This seems to only shift the header dates without adjusting the actual Calendar. June starts on Saturday, not Friday.
I like the idea though.

<img width="399" alt="image" src="https://github.com/zadam/trilium/assets/25670035/2e13085b-3726-4fb6-85c6-9a2641b0a676">

**Reply by ReinforceZwei** [reply]
The script shifts the header and the dates as well. You may want to reload the frontend to see if it works for you.
![image](https://github.com/zadam/trilium/assets/39955851/a76d5d56-2735-49a2-909c-9b6b4ef2b852)

**Comment by TheBig-O** [comment]
The screenshot I sent was after I reloaded the frontend multiple times and restarted the application as well. Regardless of what I do, the dates are off.

**Reply by ReinforceZwei** [reply]
what is your trilium version, and what browser you use?

**Reply by TheBig-O** [reply]
I'm running this in the MacOS application, version: | 0.63.6
Even in Chrome, I show the same results.
<img width="449" alt="image" src="https://github.com/zadam/trilium/assets/25670035/e0efd2fe-d118-440d-9c77-bf35c994dd1d">

**Reply by ReinforceZwei** [reply]
I have tested version 0.63.6 on Windows with Chrome and the calendar works as expected. Unfortunately I don't have a Mac. Maybe you can open the Dev Tool to see if there is any errors in the console?

**Reply by TheBig-O** [reply]
I looked in Dev Tools and didn't see any errors. Reloaded front end a couple of times and everything seems to process without errors but the numbers are still off by one day.

**Comment by Nriver** [comment]
This happens if open calendar as soon as I reload the frontend. I can see the `SUN`( or `日` in chinese version) shifts. The second time I open the calendar, then the date number shifts to the right place. Watch for how the `5` changes.

![1](https://github.com/zadam/trilium/assets/6752679/ed092e66-5e35-400c-920d-b12914be2f02)

If I wait for several seconds after the reload, that won't happen. Everything will shift to the right place.

**Comment by Nriver** [comment]
Added to https://github.com/Nriver/awesome-trilium

**Comment by meichthys** [comment]
This should probably be configurable in TriliumNext options. 
I'll create an issue to track this.
